media/videostitcher/stitcher_test.go

```diff
@@ -0,0 +1,146 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stitcher
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+	cloudresourcemanager "google.golang.org/api/cloudresourcemanager/v1"
+)
+
+const (
+	deleteCdnKeyResponse = "Deleted CDN key"
+	deleteSlateResponse  = "Deleted slate"
+	location             = "us-central1"
+	slateID              = "my-go-test-slate"
+)
+
+// To run the tests, do the following:
+// Export the following env vars:
+// *   GOOGLE_APPLICATION_CREDENTIALS
+// *   GOLANG_SAMPLES_PROJECT_ID
+// Enable the following API on the test project:
+// *   Video Stitcher API
+// *   Cloud Resource Manager API (needed for project number translation)
+
+// TestStitcher tests major operations on slates, CDN keys, VOD sessions, and
+// live sessions.
+func TestStitcher(t *testing.T) {
+	tc := testutil.SystemTest(t)
+	ctx := context.Background()
+
+	bucketName := "cloud-samples-data/media/"
+	slateURI := "https://storage.googleapis.com/" + bucketName + "ForBiggerEscapes.mp4"
+	updatedSlateURI := "https://storage.googleapis.com/" + bucketName + "ForBiggerJoyrides.mp4"
+
+	// Get the project number
+	cloudresourcemanagerClient, err := cloudresourcemanager.NewService(ctx)
+	if err != nil {
+		t.Fatalf("cloudresourcemanager.NewService: %v", err)
+	}
+	project, err := cloudresourcemanagerClient.Projects.Get(tc.ProjectID).Do()
+	if err != nil {
+		t.Fatalf("cloudresourcemanagerClient.Projects.Get.Do: %v", err)
+	}
+	projectNumber := strconv.FormatInt(project.ProjectNumber, 10)
+
+	testSlates(t, projectNumber, slateURI, updatedSlateURI)
+	t.Logf("\ntestSlates() completed\n")
+}
+
+// testSlates tests major operations on slates. Create, list, update,
+// and get operations check if the slate resource name is returned. The
+// delete operation checks for a hard-coded string response.
+func testSlates(t *testing.T, projectNumber, slateURI, updatedSlateURI string) {
```

Comment:
It's a little confusing to have a separate function for this set of tests. You can just have multiple tests in this file instead. Any shared setup between this and future tests can go in a TestMain. https://pkg.go.dev/testing#hdr-Main

Reply:
Done